src/repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java

```diff
@@ -407,82 +585,202 @@ public Action getProjectAction(AbstractProject<?, ?> project) {
+            if(useSonarQubeRepair) {
+                FormValidation.warning("Please also provide sonarRules in the textfield below");
+            }
             return FormValidation.ok();
         }
 
         public boolean isApplicable(Class<? extends AbstractProject> aClass) {
-            // Indicates that this builder can be used with all kinds of project types 
             return true;
         }
 
-        /**
-         * This human readable name is used in the configuration screen.
-         */
         public String getDisplayName() {
             return "Run repairnator";
         }
 
         @Override
         public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
-            // To persist global configuration information,
-            // set that to properties and call save().
             req.bindJSON(this, formData);
-            useNPEFix = formData.getBoolean("useNPEFix");
-            useAstorJKali = formData.getBoolean("useAstorJKali");
-            useAstorJMut = formData.getBoolean("useAstorJMut");
-            useNPEFixSafe = formData.getBoolean("useNPEFixSafe");
-            useNopolTestExclusionStrategy = formData.getBoolean("useNopolTestExclusionStrategy");
-            notifyTo = formData.getString("notifyTo");
-            // ^Can also use req.bindJSON(this, formData);
-            //  (easier when there are many fields; need set* methods for this, like setUseNPEFix)
+            this.useNPEFix = formData.getBoolean("useNPEFix");
+            this.useAstorJKali = formData.getBoolean("useAstorJKali");
+            this.useAstorJMut = formData.getBoolean("useAstorJMut");
+            this.useNPEFixSafe = formData.getBoolean("useNPEFixSafe");
+            this.useNopolTestExclusionStrategy = formData.getBoolean("useNopolTestExclusionStrategy");
+            this.useSonarQubeRepair = formData.getBoolean("useSonarQubeRepair");
+
             save();
             return true;
         }
 
-        public boolean useNPEFix() {
+        public boolean getUseNPEFix() {
             return useNPEFix;
         }
 
-        public boolean useAstorJKali() {
+        public boolean getUseAstorJKali() {
             return useAstorJKali;
         }
 
-        public boolean useAstorJMut() {
+        public boolean getUseAstorJMut() {
             return useAstorJMut;
         }
 
-        public boolean useNPEFixSafe() {
+        public boolean getUseNPEFixSafe() {
             return useNPEFixSafe;
         }
 
-        public boolean useNopolTestExclusionStrategy() {
+        public boolean getUseNopolTestExclusionStrategy() {
             return useNopolTestExclusionStrategy;
+        }  
+
+        public boolean getUseSonarQubeRepair() {
+            return useSonarQubeRepair;
+        }
+    }
+
+    public static final class SonarRulesBlock {
```

Comment:
for sake of encapsulation, avoid public inner classes, either private inner or public top-level.

Reply:
Ack. I'll make this private and see if it works for the UI.

Reply:
![Screenshot from 2020-06-01 21-14-37](https://user-images.githubusercontent.com/33935517/83445255-f4d08f00-a44c-11ea-8929-43ecc7a9c8fb.png)

It does not seems to be possible to change to private without getting `java.lang.IllegalAccessException`  . I can't move it either because this thing need to be a subclass to be understood as  a UI subcomponent of the main UI, otherwise the `.jelly` can't find it. 

Documentation about how to build these are very bad, I got this working mostly by piecing together the scrap info on multiple Stackoverflow pages.